pkg/cmd/server/kubernetes/master/master_config.go

```diff
@@ -169,6 +170,7 @@ func BuildStorageFactory(masterConfig configapi.MasterConfig, server *kapiserver
 	if err != nil {
 		return nil, err
 	}
+	storageGroupsToEncodingVersion[autoscaling.GroupName] = kubecompatextensionsv1beta1.SchemeGroupVersion
```

Comment:
Existing 3.5 clusters have existing HPAs encoded as extensions/v1beta1. Unless we make sure customers do a migration in 3.5.x, we have to keep them in extensions/v1beta1.